src/Pass/ImgTagTransformPass.php

```diff
@@ -66,33 +66,49 @@ function pass()
         foreach ($all_a as $el) {
             /** @var \DOMElement $dom_el */
             $dom_el = $el->get(0);
-            $lineno = $this->getLineNo($dom_el);
             if ($this->isSvg($dom_el)) {
                 // @TODO This should be marked as a validation warning later?
                 continue;
             }
+            $lineno = $this->getLineNo($dom_el);
             $context_string = $this->getContextString($dom_el);
-            $new_dom_el = $this->cloneAndRenameDomElement($dom_el, 'amp-img');
-            $new_el = $el->prev();
-
-            $success = $this->setResponsiveImgHeightAndWidth($new_el);
-            // We were not able to get the image dimensions, abort conversion.
-            if (!$success) {
+            $hasWidthAndHeight = $this->setResponsiveImgHeightAndWidth($el);
+            if (!$hasWidthAndHeight) {
                 $this->addActionTaken(new ActionTakenLine('img', ActionTakenType::IMG_COULD_NOT_BE_CONVERTED, $lineno, $context_string));
-                // Abort the conversion and remove the new img tag
-                $new_el->remove();
                 continue;
             }
-
+            if ($this->isPixel($el)) {
+                $new_dom_el = $this->convertAmpPixel($el);
+                $this->context->addLineAssociation($new_dom_el, $lineno);
```

Comment:
remove `$this->context->addLineAssociation($new_dom_el, $lineno);` from the if branches and do it once before `$el->remove();`